src/Raven.Client/Documents/Operations/Operation.cs

```diff
@@ -135,6 +136,23 @@ protected virtual async Task Process()
             }
         }
 
+        private void OnConnectionStatusChanged(object sender, EventArgs e)
+        {
+            if(e is DatabaseChanges.OnReconnect)
+                AsyncHelpers.RunSync(OnConnectionStatusChangedAsync);
+        }
+
+        private async Task OnConnectionStatusChangedAsync()
+        {
+            try
+            {
+                await FetchOperationStatus(shouldThrow: false).ConfigureAwait(false);
```

Comment:
do we need all of the retires for this case? we need to check it only once right?

Reply:
Probably one time is enough but I wanted to avoid making a dedicated function that would look very similar or passing too many parameters